pkg/identity/cache/cache.go

```diff
@@ -48,23 +48,28 @@ func (s IdentitiesModel) Less(i, j int) bool {
 func GetIdentityCache() IdentityCache {
 	cache := IdentityCache{}
 
-	IdentityAllocator.ForeachCache(func(id idpool.ID, val allocator.AllocatorKey) {
-		if val != nil {
-			if gi, ok := val.(globalIdentity); ok {
-				cache[identity.NumericIdentity(id)] = gi.LabelArray()
-			} else {
-				log.Warningf("Ignoring unknown identity type '%s': %+v",
-					reflect.TypeOf(val), val)
+	if IdentityAllocator != nil {
```

Comment:
This change is fine, but I'm wondering what the motivation for it is - is it for easier unit testing? Optimization in certain cases?

Reply:
`GetIdentityCache()` required to call `InitIdentityAllocator()` beforehand so far. This is always true for non-tests. `GetIdentityCache()` is reasonable to call even without an allocator in which case it will return all reserved identities. This is what the unit test does and requires.

Reply:
@aanm I'm not sure I understand the question. How is the mutex related to this change?

Reply:
There is a mutex protecting this variable as the variable is nil and then is not nil. The mutex does not seem to protect this variable for all code paths.

Reply:
Are you talking about this mutex?
```
        // setupMutex synchronizes InitIdentityAllocator() and Close()
        setupMutex lock.Mutex
```

The comment says that it is used to synchronize Init() and Close(). What is the relation to this PR. This PR is adding a unit test and requires this change by not wanting to initialize the identity allocator. That's all.

Reply:
Why only the `Init` and `Close()`? What about all the other reads performed on that variable?

Reply:
@aanm Based on our offline discussion, what is your conclusion here?